src/setSalt.ts

```diff
@@ -0,0 +1,5 @@
+import { hash } from 'fictional'
+
+export const setSalt = (nextSalt: string) => {
+  ;(hash as any).salt = nextSalt
```

Comment:
What does the ; do?

Reply:
Good question :)

It was added automatically by prettier in vscode when I added `(hash as any)`.

The `;` does "nothing" technically. I can't find a good link now, but iirc it goes something like: when using no semicolons with js, its a convention people adopted to avoid ambiguity in statements that start with a `(` or `[`.  For example:

```js
foo
(bar)
```

Should this be interpreted as `foo(bar)` or as `foo; bar;`? In most cases, [js can figure out where to put semicolons](https://en.wikibooks.org/wiki/JavaScript/Automatic_semicolon_insertion), but cases like these are an exception. So people settled on a convention where statements starting with `(` and `[` are prefixed with a `;` (which does "nothing", but allows us to disambiguate). And so prettier adopted that convention for the semicolonless style. Something like that.

Reply:
That actually makes sense! Thank you!